src/main/java/sx/blah/discord/api/internal/Requests.java

```diff
@@ -278,24 +276,26 @@ private String request(HttpUriRequest request) throws DiscordException, RateLimi
 
 				if (responseCode == 404) {
 					if (!request.getURI().toString().contains("invite") && !request.getURI().toString().contains("messages")) //Suppresses common 404s which are a result on queries to verify if something exists or not
-					    LOGGER.error(LogMarkers.API, "Received 404 error, please notify the developer and include the URL ({})", request.getURI());
+						LOGGER.error(LogMarkers.API, "Received 404 error, please notify the developer and include the URL ({})", request.getURI());
 					return null;
 				} else if (responseCode == 403) {
 					LOGGER.error(LogMarkers.API, "Received 403 forbidden error for url {}. If you believe this is a Discord4J error, report this!", request.getURI());
 					return null;
 				} else if (responseCode == 204) { //There is a no content response when deleting messages
 					return null;
-				} else if (responseCode == 502) {
-					LOGGER.trace(LogMarkers.API, "502 response on request to {}, response text: {}", request.getURI(), message); //This can be used to verify if it was cloudflare causing the 502.
-
-					if (message.toLowerCase(Locale.ROOT).contains("cloudflare")) {
-						throw new DiscordException("502 error on request to " + request.getURI()
-								+ ". This is due to CloudFlare.");
+				} else if (responseCode >= 500 && responseCode < 600) {
```

Comment:
Just interjecting with comments with from b1zny: https://puu.sh/t97G9/de0c1e736c.png and https://puu.sh/t97Ki/74f237acac.png